internal/api/user_devices.go

```diff
@@ -321,6 +321,12 @@ func (s *userDeviceService) deviceModelToAPI(ud *models.UserDevice) *pb.UserDevi
 
 		if amnft := vnft.R.VehicleTokenAutopiUnit; amnft != nil {
 			out.AftermarketDeviceTokenId = s.toUint64(amnft.TokenID)
+
+			if vnft.R.VehicleTokenAutopiUnit.Beneficiary.Valid {
```

Comment:
Note that you can use `amnft` instead of `vnft.R.VehicleTokenAutopiUnit` around here.